blob/blob.go

```diff
@@ -91,3 +132,47 @@ func NewBlob(shareVersion uint8, namespace namespace.ID, data []byte) (*Blob, er
 func (b *Blob) Namespace() namespace.ID {
 	return append([]byte{uint8(b.NamespaceVersion)}, b.NamespaceId...)
 }
+
+func (b *Blob) MarshalJSON() ([]byte, error) {
+	appBlob := &types.Blob{
+		NamespaceId:      b.Namespace(),
+		Data:             b.Data,
+		ShareVersion:     b.ShareVersion,
+		NamespaceVersion: b.NamespaceVersion,
+	}
+	data, err := tmjson.Marshal(appBlob)
+	if err != nil {
+		return nil, err
+	}
+
+	return json.Marshal(&struct {
+		Blob       json.RawMessage `json:"blob"`
+		Commitment []byte          `json:"commitment"`
+	}{
+		Blob:       data,
+		Commitment: b.Commitment,
+	},
```

Comment:
Dedup as private type?

Reply:
I would also recommend having Blob JSON marshaling without the inner Blob.
So it's not Blob{blob: {}}. 
We will eventually make Blob one single struct without embedding the app's blob. Once we come to that point, it would be super weird to have Blob{blob: {}}, and changing that would be breaking, so let's make it correct from the beginning.

We could also avoid importing tmjson this way (?)